internal/httputil/httputils.go

```diff
@@ -59,7 +59,7 @@ func BuildParams(params interface{}, noUrlEncode ...bool) (encodedParamStr strin
 			encodedParamStr += "&"
 		}
 		s = gconv.String(v)
-		if urlEncode && len(s) > 6 && strings.Compare(s[0:6], fileUploadingKey) != 0 {
+		if urlEncode && (len(s) <= 6 || strings.Compare(s[0:6], fileUploadingKey) != 0) {
 			s = gurl.Encode(s)
```

Comment:
It may be:
```go
if urlEncode {
	if len(s) > 6 && strings.Compare(s[0:6], fileUploadingKey) == 0 {
		// No url encoding if uploading file.
	} else {
		s = gurl.Encode(s)
	}
}
```

Reply:
🆗  I just want to keep its original style in my first commit, but I agree with u more : )